wolfcrypt/src/tfm.c

```diff
@@ -711,11 +762,10 @@ void fp_div_2d(fp_int *a, int b, fp_int *c, fp_int *d)
     return;
   }
 
-  fp_init(&t);
-
   /* get the remainder */
   if (d != NULL) {
-    fp_mod_2d (a, b, &t);
+    /* NOTE: d must not be the same pointer as a or b */
```

Comment:
Should we be checking the comment assumption here?

Reply:
Fixed.
Using logic I was able to eliminate the problem,